src/math/p5.Vector.js

```diff
@@ -2138,6 +2151,34 @@ p5.Vector.mult = function mult(v, n, target) {
   return target;
 };
 
+/**
+ * Rotates the vector (only 2D vectors) by the given angle, magnitude remains the same and returns a new vector.
+ */
+
+/**
+ * @method rotate
+ * @static
+ * @param  {p5.Vector} v
+ * @param  {Number} angle
+ * @param  {p5.Vector} [target] the vector to receive the result (Optional)
+ */
+
+p5.Vector.rotate = function rotate(v, a, target) {
+  if (!target) {
+    target = v.copy();
+    if (arguments.length === 3) {
+      p5._friendlyError(
+        'The target parameter is undefined, it should be of type p5.Vector',
+        'p5.Vector.rotate'
+      );
```

Comment:
This friendly error message is a little confusing to me, can you clarify how this works? If you've just copied `target` from `v` above, wouldn't it be defined then?